traefik/templates/servicemonitor.yaml

```diff
@@ -11,6 +11,8 @@ metadata:
   name: {{ template "traefik.fullname" . }}
   {{- with .Values.metrics.prometheus.serviceMonitor.namespace }}
   namespace: {{ . }}
+  {{- else }}
+  namespace: {{ template "traefik.namespace" . }}
   {{- end }}
```

Comment:
`with ... else ... end` does not make sense. [with](https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with) is for setting scope,

Reply:
Actually I understand that it seems less readable but it works. According to the documentation, `with` works exactly like `if` but it also get the scope, so `with ... else` is completely valid. This is also attested by tests. Apart from technicalities though, I understand it is not very readable so it's ok to change it

Reply:
thanks